code/cfile/cfile.cpp

```diff
@@ -1120,6 +1126,45 @@ int cf_get_dir_type(CFILE *cfile)
 	return Cfile_block_list[cfile->id].dir_type;
 }
 
+/**
+ * Stuffs Pathtypes with an additional path leading to a subdir of the same name as the given file (minus file extension).
+ * Currently used for allowing frames of .eff animations to be found in their own subdirectories.
+ *
+ * Example: if called with "debris.eff" and that file is in data/maps, then Pathtypes[CF_TYPE_TEMP_SUBDIR_LOOKUP].path
+ * will be stuffed with data/maps/debris, allowing data/maps/debris/debris_0000.dds (etc) to be found
+ */
+bool cf_set_temp_subdir_pathtype(const char *filename) {
+	mprintf(("CFILE: Setting temporary pathtype for %s... ", filename));
+
+	// Finds the relative path of where the file exists and appends the filename
+	// (without extension) to it, giving us a relative path that can be temporarily
+	// stuffed into Pathtypes
+	for (size_t i=CF_TYPE_ROOT; i<CF_MAX_PATH_TYPES; i++) {
+		size_t size, offset; // Needed for cf_find_file_location, but not used
+
+		if (cf_find_file_location(filename, i, 0, NULL, &size, &offset)) {
+			SCP_string subdir, finalpath;
+
+			subdir = SCP_string(filename);
+			subdir = subdir.substr(0, subdir.find_last_of("."));
```

Comment:
If filename doesn't contain an extension then will `find_last_of` will return a value that is not valid for `substr`.